pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorConfig.java

```diff
@@ -0,0 +1,181 @@
+    _partitioningConfig = partitioningConfig;
+    _collectorConfig = collectorConfig;
+    _segmentConfig = segmentConfig;
+  }
+
+  /**
+   * The Pinot table config
+   */
+  public TableConfig getTableConfig() {
+    return _tableConfig;
+  }
+
+  /**
+   * The Pinot schema
+   */
+  public Schema getSchema() {
+    return _schema;
+  }
+
+  /**
+   * The RecordTransformerConfig for the SegmentProcessorFramework's map phase
+   */
+  public RecordTransformerConfig getRecordTransformerConfig() {
+    return _recordTransformerConfig;
+  }
+
+  /**
+   * The RecordFilterConfig to filter records
+   */
+  public RecordFilterConfig getRecordFilterConfig() {
+    return _recordFilterConfig;
+  }
+
+  /**
+   * The PartitioningConfig for the SegmentProcessorFramework's map phase
+   */
+  public PartitioningConfig getPartitioningConfig() {
+    return _partitioningConfig;
+  }
+
+  /**
+   * The CollectorConfig for the SegmentProcessorFramework's reduce phase
+   */
+  public CollectorConfig getCollectorConfig() {
+    return _collectorConfig;
+  }
+
+  /**
+   * The SegmentConfig for the SegmentProcessorFramework's segment generation phase
+   */
+  public SegmentConfig getSegmentConfig() {
+    return _segmentConfig;
+  }
+
+  /**
+   * Builder for SegmentProcessorConfig
+   */
+  public static class Builder {
+    private TableConfig tableConfig;
+    private Schema schema;
+    private RecordTransformerConfig recordTransformerConfig;
+    private RecordFilterConfig recordFilterConfig;
+    private PartitioningConfig partitioningConfig;
+    private CollectorConfig collectorConfig;
+    private SegmentConfig _segmentConfig;
+
+    public Builder setTableConfig(TableConfig tableConfig) {
+      this.tableConfig = tableConfig;
+      return this;
+    }
+
+    public Builder setSchema(Schema schema) {
+      this.schema = schema;
+      return this;
+    }
+
+    public Builder setRecordTransformerConfig(RecordTransformerConfig recordTransformerConfig) {
+      this.recordTransformerConfig = recordTransformerConfig;
+      return this;
+    }
+
+    public Builder setRecordFilterConfig(RecordFilterConfig recordFilterConfig) {
+      this.recordFilterConfig = recordFilterConfig;
+      return this;
+    }
+
+    public Builder setPartitioningConfig(PartitioningConfig partitioningConfig) {
+      this.partitioningConfig = partitioningConfig;
+      return this;
+    }
+
+    public Builder setCollectorConfig(CollectorConfig collectorConfig) {
+      this.collectorConfig = collectorConfig;
+      return this;
+    }
+
+    public Builder setSegmentConfig(SegmentConfig segmentConfig) {
+      this._segmentConfig = segmentConfig;
+      return this;
+    }
+
+    public SegmentProcessorConfig build() {
+      Preconditions.checkNotNull(tableConfig, "Must provide table config in SegmentProcessorConfig");
+      Preconditions.checkNotNull(schema, "Must provide schema in SegmentProcessorConfig");
+      if (recordTransformerConfig == null) {
```

Comment:
Suggest leaving them as `null` if not set

Reply:
i just wanted to avoid checking null before using them in the mapper and reducer